packages/hix/lib/Hix/Managed/Data/StateFileConfig.hs

```diff
@@ -1,12 +1,14 @@
 module Hix.Managed.Data.StateFileConfig where
 
-import Path (File, Path, Rel, relfile)
+import Path (File, SomeBase (Rel), relfile)
+
 import Distribution.Pretty (Pretty (pretty))
+import Hix.Data.PathSpec (PathSpec (PathConcrete))
 import Hix.Pretty (prettyMap, field)
 
 data StateFileConfig =
   StateFileConfig {
-    file :: Path Rel File
+    file :: PathSpec File
```

Comment:
As discussed in the other thread: this may only be a relative path, and it's not related to cwd

Reply:
If we force a relative path we wouldn't have the resolving for this tough (like handling `..`). Is that okay? In the usage it works as expected with this but I guess we don't fail fast if an absolute path is passed, we want that?

Reply:
yes, it should not be allowed to be resolved. This is a path relative to the project root that is configured in Nix (see [here](https://github.com/tek/hix/blob/main/modules/managed.nix#L31)); neither parent directories nor absolute paths would make any sense for it.

Reply:
alright yes, makes sense. Updated it back to relative path only.